upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -77,6 +77,10 @@ func (tf *TemplateFunctions) AddTo(dest template.FuncMap, secretStore fi.SecretS
 	dest["join"] = func(a []string, sep string) string {
 		return strings.Join(a, sep)
 	}
+	dest["indent"] = func(spaces int, v string) string {
```

Comment:
This looks pretty good and seems to do the job for me.

Rather than copying the function, would you mind referencing it as we do for our templater in the toolbox?

https://github.com/kubernetes/kops/blob/298f79659a2bd9c1d6548c6c95b335b5e29e6466/pkg/util/templater/templater.go#L76-L92

Reply:
My initial approach was this:

```diff
diff --git pkg/templates/templates.go pkg/templates/templates.go
index 8c92208fb..78e31e5fe 100644
--- pkg/templates/templates.go
+++ pkg/templates/templates.go
@@ -28,6 +28,8 @@ import (
        "k8s.io/kops/pkg/apis/kops"
        "k8s.io/kops/upup/pkg/fi"
        "k8s.io/kops/util/pkg/vfs"
+
+       "github.com/Masterminds/sprig"
 )

 type Templates struct {
@@ -98,7 +100,7 @@ func (t *Templates) loadFrom(base vfs.Path) error {
 func (l *Templates) executeTemplate(key string, d string) (string, error) {
        t := template.New(key)

-       funcMap := make(template.FuncMap)
+       funcMap := sprig.TxtFuncMap()
        //funcMap["Args"] = func() []string {
        //      return args
        //}
```

which fixes the issue. But then I saw `upup/pkg/fi/cloudup.TemplateFunctions.AddTo` and thought it would be too out of scope to replace the base `template.FuncMap` for the main templates package.

Also, there is a few places adding/replacing `join`, `replace` and other conflicting functions.

If you agree this would be too much out of the scope, another solution could be copying just the `indent` function, like:

```diff
diff --git upup/pkg/fi/cloudup/template_functions.go upup/pkg/fi/cloudup/template_functions.go
index f446a9f0c..cda8ae6c0 100644
--- upup/pkg/fi/cloudup/template_functions.go
+++ upup/pkg/fi/cloudup/template_functions.go
@@ -77,10 +77,9 @@ func (tf *TemplateFunctions) AddTo(dest template.FuncMap, secretStore fi.SecretS
        dest["join"] = func(a []string, sep string) string {
                return strings.Join(a, sep)
        }
+       txtFuncMap := sprig.TxtFuncMap()
+       dest["indent"] = txtFuncMap["indent"]
```



Also, the snippet you sent replaces `sprig.TxtFuncMap()["indent"]` function with another implementation?

Reply:
Ahh yes great points. Though I'd love to do the first suggestion, since this will likely be cherry picked into 1.15 pretty soon, I'd rather do the smaller scope fix (second).  Anyway to reuse code from other repos, embracing their licenses makes life much better.

Thanks so much

Reply:
Done.